examples/navigation-parcel/client.tsx

```diff
@@ -0,0 +1,4 @@
+'use client-entry';
+import {hydrate} from '@parcel/rsc/client';
+
+hydrate();
```

Comment:
If you have a way to trigger a re-fetch of the RSC payload, you can implement `onHmrRefresh` here to trigger that. Then when you edit a server component it will update without losing client only state.

Reply:
@devongovett Thanks for tracking down the bug. I've updated navigation-react to fix the PR.

Also, thanks for pointing out `onHmrRefresh`. I think I can do something good here. The stateNavigator routes might've changed so the Navigation router has to rerender all links as well.

I think parcel rsc only supports esm for npm packages that contain React components. I had to switch navigation-react to esm although I left navigation in cjs. Is this something you know about?

Reply:
> I think parcel rsc only supports esm for npm packages that contain React components.

What was the issue that you were running into?

Reply:
Browse node_modules to the package.json of navigation-react and [delete the esm export lines](https://github.com/grahammendick/navigation/blob/rsc-web/build/npm/navigation-react/package.json#L25-L36). Then if you run `npm run build` of the navigation-parcel example you'll get an error.

But if you had [deleted the cjs export lines](https://github.com/grahammendick/navigation/blob/rsc-web/build/npm/navigation-react/package.json#L13-L24) instead then you could build the navigation-parcel no problem.

Parcel only works with esm exports for navigation-react.  But Parcel works with cjs for the other node_module dependencies, for example, the navigation package is cjs. So I'm guessing it's something to do with packages that export React components?

Reply:
I got this [same example working using webpack](https://github.com/grahammendick/navigation/tree/rsc-web/NavigationReact/sample/rsc-webpack) and it also failed when I deleted the esm export lines. So I'm guessing that RSC bundlers don't like 3rd-party cjs component libraries that have different 'react-server' exports (or I'm doing something wrong).